### PR TITLE
Add newer asset loading and more up to date classes

### PR DESCRIPTION
**Pull request by joshfeck**
This was originally done in a branch as part of https://events.codebasehq.com/projects/event-espresso/tickets/10381

It kind of fell by the wayside, but we want to start extracting some of the logic out of the templates before adding any more new logic (and possibly add some new templates).

So I brought this over to GitHub and applied a few fixes that have been added to master since this branch was first cut.

## Checklist
* [x] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [x] User input is adequately validated and sanitized
* [x] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [x] My code is tested.
* [x] My code follows the Event Espresso code style.
* [x] My code has proper inline documentation.
* [x] My code accounts for when the site is in Maintenance Mode (MM2 especially disallows usage of models)

### CODE REVIEW

File: domain/entities/shortcodes/EventsTableTemplate.php [hunk 1]
```diff
@@ -0,0 +1,301 @@
+<?php
+
+namespace EventEspresso\TableTemplate\domain\entities\shortcodes;
+
+use EEH_Template;
+use EventEspresso\core\services\shortcodes\EspressoShortcode;
+use EventEspresso\TableTemplate\domain\queries\EventsTableTemplateQuery;
+
+defined('EVENT_ESPRESSO_VERSION') || exit;
```
Comment (tn3rb) [inline domain/entities/shortcodes/EventsTableTemplate.php]: this needs to be removed to be PSR compliant

File: EE_Events_Table_Template.class.php [hunk 2]
```diff
@@ -1,88 +1,173 @@
...
+use EventEspresso\core\exceptions\InvalidInterfaceException;
+use EventEspresso\core\services\collections\CollectionInterface;
+use EventEspresso\core\services\loaders\Loader;
+use EventEspresso\core\services\loaders\LoaderFactory;
+use EventEspresso\core\services\loaders\LoaderInterface;
+use EventEspresso\core\services\shortcodes\ShortcodeInterface;
+
+if (! defined('EVENT_ESPRESSO_VERSION')) {
+    exit();
+}
```
Comment (tn3rb) [inline EE_Events_Table_Template.class.php]: this should be removed in order to be PSR compliant

File: domain/entities/shortcodes/EventsTableTemplate.php [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+    protected function customAttributeSanitizationMap()
+    {
+        return array(
+            'category_slug' => 'skip_sanitization',
+            'show_expired'  => 'skip_sanitization',
+            'order_by'      => 'skip_sanitization',
+            'month'         => 'skip_sanitization',
+            'sort'          => 'skip_sanitization',
+        );
+    }
```
Comment (tn3rb) [inline domain/entities/shortcodes/EventsTableTemplate.php]: sure we don't want to sanitize some of these?

ie:

```php
'category_slug' => 'sanitize_text_field',
```

Comment (joshfeck) [reply domain/entities/shortcodes/EventsTableTemplate.php]: They already get sanitized/whitelisted in EEH_Event_Query.

**Pull request closed without merging**